apps/omg/lib/omg/state.ex

```diff
@@ -129,7 +129,13 @@ defmodule OMG.State do
   def handle_info(:send_metrics, state) do
     _ =
       Core.Metrics.calculate(state)
-      |> Enum.map(fn {key, value} -> OMG.Utils.Metrics.gauge(key, value) end)
+      |> Enum.map(fn
```

Comment:
I think it would be better if we completely separate logic from metrics and use a more potterhat approach: https://github.com/omisego/potterhat/pull/29/files#diff-eb43df535cf2c41479f62958916f13f3R15

This means we setup a separate handler (we could have handler per application, for example) and do any sort of data manipulation (`Core.Metrics.calculate(state)`) there.

Reply:
you suggest to use `:sys.get_state/1`?
I see hear little problems if we use this approach, there will be making a copy of a state and this can be problematic.

Or You think about making :telemetry.execute with state as one of data?
this will have less impact, but it will be a bit inconsistent with encapsulation.

Reply:
just to be clear - no, I did not mean we should use `:sys.get_state`.

I think we should pass :telemetry.execute the state, but have a handler just for `OMG.State`

Reply:
my mistake thought that by `completely separate logic from metrics`, you mean even remove from logic :telemetry.execute

Will not it be a little bit misleading? 
Seeing something like this, I would have the impression that someone else is using it.
I know from experience that later will appear new handlers, even if they shouldn't.

Reply:
I will look at the potterhead again, 
maybe the code can be simplified.

Reply:
The server uses Metric_Calc_Server to calculate metrics, result sent by telemetry.
TelemetryMetricsStatsd deals with the rest

example: additional TelemetryMetricsStatsd code in supervisor
``` elixir 
{TelemetryMetricsStatsd,
       [
         metrics: [
           counter("child_chain.deposits.length", event_name: "depositor.events"),
           last_value("child_chain.unique_users.value", event_name: "Elixir.OMG.State.unique_users"),
           last_value("child_chain.balance.value", event_name: "Elixir.OMG.State.balance", tags: [:coin])
         ],
         formatter: :datadog
       ]}
```
- easy testing of counting metrics
- easily visible sendings metrics (supervisor)
- it does not make the whole state available through telemetry

===============================================================================

Server sends state by telemetry
subscribed Potterhat calculate metrics and send them using Statsd

- can also create separate Potterhat for each server (it is not created God Class)
- a bit harder to test `cauting metrics` (every handler must send metrick by Statsd)


Did I forget about something?
(of course, any of them don't do large copy :D )

Reply:
There’s no copying in Telemetry because it’s in the same process context.